src/utils/publishToNpm.js

```diff
@@ -41,24 +41,47 @@ async function allowNpmPublish(version) {
 
   return !packageVersionInfo
 }
-
+/**
+ * 
+ * @param {
+ *  npmToken,
+    opticToken,
+    opticUrl,
+    npmTag,
+    version,
+    provenance: boolean
+    hasAccess: boolean
+ * }  
+ */
 async function publishToNpm({
   npmToken,
   opticToken,
   opticUrl,
   npmTag,
   version,
   provenance,
+  hasAccess
 }) {
   await execWithOutput('npm', [
     'config',
     'set',
     `//registry.npmjs.org/:_authToken=${npmToken}`,
   ])
 
+  let packageName = null
+  try {
+    const packageInfo = await execWithOutput('npm', ['view', '--json'])
+    packageName = packageInfo ? JSON.parse(packageInfo).name : null
+  } catch (error) {
+    if (!error?.message?.match(/code E404/)) {
```

Comment:
Nit, but I wonder if we can avoid duplicating this, e.g. have the function that calls it already return a status number. or pull it out into its own utility that passes the "is published" state into `allowNpmPublish`?